pipeline/ephys.py

```diff
@@ -0,0 +1,282 @@
+import datajoint as dj
+
+from . import lab, experiment
+from . import get_schema_name
+
+import numpy as np
+from scipy.interpolate import CubicSpline
+
+schema = dj.schema(get_schema_name('ephys'))
+
+
+@schema
+class ProbeInsertion(dj.Manual):
+    definition = """
+    -> experiment.Session
+    insertion_number: int
+    ---
+    -> lab.Probe
+    -> lab.ElectrodeConfig
+    """
+
+    class InsertionLocation(dj.Part):
+        definition = """
+        -> master
+        ---
+        -> lab.SkullReference
+        ap_location: decimal(6, 2) # (um) anterior-posterior; ref is 0; more anterior is more positive
+        ml_location: decimal(6, 2) # (um) medial axis; ref is 0 ; more right is more positive
+        depth:       decimal(6, 2) # (um) manipulator depth relative to surface of the brain (0); more ventral is more negative
+        theta:       decimal(5, 2) # (deg) - elevation - rotation about the ml-axis [0, 180] - w.r.t the z+ axis
+        phi:         decimal(5, 2) # (deg) - azimuth - rotation about the dv-axis [0, 360] - w.r.t the x+ axis
+        beta:        decimal(5, 2) # (deg) rotation about the shank of the probe [-180, 180] - clockwise is increasing in degree - 0 is the probe-front facing anterior
+        """
+
+    class RecordableBrainRegion(dj.Part):
+        definition = """
+        -> master
+        -> lab.BrainArea
+        -> lab.Hemisphere
+        """
+
+    class InsertionNote(dj.Part):
+        definition = """
+        -> master
+        ---
+        insertion_note: varchar(1000)
+        """
+
+    class RecordingSystemSetup(dj.Part):
+        definition = """
+        -> master
+        ---
+        sampling_rate: int  # (Hz)
+        """
+
+
+@schema
+class LFP(dj.Imported):
+    definition = """
+    -> ProbeInsertion
+    ---
+    lfp_sample_rate: float          # (Hz)
+    lfp_time_stamps: longblob       # timestamps with respect to the start of the recording (recording_timestamp)
+    lfp_mean: longblob              # mean of LFP across electrodes
+    """
+
+    class Channel(dj.Part):
+        definition = """  
+        -> master
+        -> lab.ElectrodeConfig.Electrode
+        ---
+        lfp: longblob           # recorded lfp at this electrode
+        """
+
+
+@schema
+class UnitQualityType(dj.Lookup):
+    definition = """
+    # Quality
+    unit_quality  :  varchar(100)
+    ---
+    unit_quality_description :  varchar(4000)
+    """
+    contents = [
+        ('good', 'single unit'),
+        ('ok', 'probably a single unit, but could be contaminated'),
+        ('multi', 'multi unit'),
+        ('all', 'all units')
+    ]
+
+
```

Comment:
We need to keep track of whether units were phototagged (responsive to light). That could go in a schema like this, maybe:
```suggestion
@schema
class Phototag(dj.Manual):
    definition = """
    -> PhotoStim
    ---
    responses  : varchar(30)   # Yes / No / Maybe
    responsive_channels= null : varchar(30)  # responsive channels
    response_delay=  null : smallint # ms
    """
```